tests/acl/test_acl_dscp_change.py

```diff
@@ -0,0 +1,657 @@
+            inner_frame=inner_packet.copy()
+        )
+    elif 'v6_in_v4' == encap_type:
+        exp_pkt = testutils.simple_vxlan_packet(
+            eth_src=duthost.facts['router_mac'],
+            ip_src=outer_ip_src,
+            ip_dst="0.0.0.0",  # We don't care about the outer dest IP
+            ip_tos=expectedDscp * 4,
+            udp_dport=setUp_vnet['vxlan_port'],
+            vxlan_vni=vxlan_vni,
+            inner_frame=inner_packet.copy()
+        )
+    elif 'v6_in_v6' == encap_type:
+        exp_pkt = testutils.simple_vxlanv6_packet(
+            eth_src=duthost.facts['router_mac'],
+            ipv6_src=outer_ip_src,
+            ipv6_dst="::",  # We don't care about the outer dest IP
+            ipv6_tc=expectedDscp * 4,
+            udp_dport=setUp_vnet['vxlan_port'],
+            vxlan_vni=vxlan_vni,
+            inner_frame=inner_packet.copy()
+        )
+    else:
+        raise ValueError(f"Unsupported encap_type: {encap_type}")
+
+    exp_pkt = mask.Mask(exp_pkt)
+    exp_pkt.set_do_not_care_scapy(Ether, "dst")
+
+    if 'in_v4' in encap_type:
+        exp_pkt.set_do_not_care_scapy(IP, "ihl")
+        exp_pkt.set_do_not_care_scapy(IP, "len")
+        exp_pkt.set_do_not_care_scapy(IP, "id")
+        exp_pkt.set_do_not_care_scapy(IP, "flags")
+        exp_pkt.set_do_not_care_scapy(IP, "frag")
+        exp_pkt.set_do_not_care_scapy(IP, "ttl")
+        exp_pkt.set_do_not_care_scapy(IP, "proto")
+        exp_pkt.set_do_not_care_scapy(IP, "chksum")
+        exp_pkt.set_do_not_care_scapy(IP, "ttl")
+        exp_pkt.set_do_not_care_scapy(IP, "dst")
+        exp_pkt.set_do_not_care_scapy(UDP, 'sport')
+        exp_pkt.set_do_not_care_scapy(UDP, 'len')
+        exp_pkt.set_do_not_care_scapy(UDP, 'chksum')
+    elif 'in_v6' in encap_type:
+        exp_pkt.set_do_not_care_scapy(IPv6, "plen")
+        exp_pkt.set_do_not_care_scapy(IPv6, "hlim")
+        exp_pkt.set_do_not_care_scapy(IPv6, "nh")
+        exp_pkt.set_do_not_care_scapy(IPv6, "dst")
+        exp_pkt.set_do_not_care_scapy(UDP, 'sport')
+        exp_pkt.set_do_not_care_scapy(UDP, 'len')
+        exp_pkt.set_do_not_care_scapy(UDP, 'chksum')
+
+    exp_pkt.set_do_not_care_scapy(VXLAN, 'flags')
+    exp_pkt.set_do_not_care_scapy(VXLAN, 'reserved1')
+    exp_pkt.set_do_not_care_scapy(VXLAN, 'reserved2')
+
+    total_size = exp_pkt.size
+    # We also dont care about the inner IP header checksum and TTL fields for both IPv4 and IPv6
+
+    if 'v4_in' in encap_type:
+        inner_ether_hdr_start = total_size - len(exp_pkt.exp_pkt[VXLAN][Ether])
+        inner_ether_hdr_end = total_size - len(exp_pkt.exp_pkt[VXLAN][IP])
+        for iter in range(inner_ether_hdr_start, inner_ether_hdr_end):
+            exp_pkt.mask[iter] = 0x00
+
+        exp_pkt.mask[inner_ether_hdr_end + 8] = 0x00  # TTL is changed
+        exp_pkt.mask[inner_ether_hdr_end + 10] = 0x00  # checksum is changed
+        exp_pkt.mask[inner_ether_hdr_end + 11] = 0x00  # checksum is changed
```

Comment:
What's the purpose to check TTL and checksum to 0?

Reply:
This part adds the packet mask. setting the mask means that this portion of the received packet should be ignored when comparing with the expected packet.